core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -971,10 +971,20 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
-  def lockWorkspace(workspaceName: WorkspaceName): Future[PerRequestMessage] =
-    dataSource.inTransaction { dataAccess =>
-      withWorkspaceContext(workspaceName, dataAccess) { workspaceContext =>
-        requireAccessIgnoreLock(workspaceContext.workspace, SamWorkspaceActions.own) {
+  def lockWorkspace(workspaceName: WorkspaceName): Future[PerRequestMessage] = {
+
+    val wsCtxFuture = dataSource.inTransaction { dataAccess =>
+      withWorkspaceContext(workspaceName, dataAccess, Some(WorkspaceAttributeSpecs(all = false))) { workspaceContext =>
+        DBIO.successful(workspaceContext)
+      }
+    }
```

Comment:
simplify this to:
```
val wsCtxFuture = getWorkspaceContext(workspaceName)
```
or inline it to the structure below:
```
    //don't do the sam REST call inside the db transaction.
    getWorkspaceContext(workspaceName) flatMap { workspaceContext =>
        requireAccessIgnoreLockF(workspaceContext.workspace, SamWorkspaceActions.own) {
            ...
```
( same comment applies to unlockWorkspace and getBucketUsage )